android_splash_screen/android/app/src/main/kotlin/com/example/splash_screen_sample/MainActivity.kt

```diff
@@ -0,0 +1,202 @@
+package com.example.splash_screen_sample
+
+import androidx.core.view.WindowCompat
+import io.flutter.embedding.android.FlutterActivity
+import android.animation.AnimatorSet
+import android.animation.ObjectAnimator
+import android.animation.ValueAnimator
+import android.os.Bundle
+import android.transition.AutoTransition
+import android.transition.TransitionManager
+import android.view.View
+import android.view.animation.AccelerateDecelerateInterpolator
+import android.widget.FrameLayout
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.constraintlayout.widget.ConstraintSet
+import androidx.core.animation.doOnEnd
+import androidx.core.splashscreen.SplashScreen.Companion.installSplashScreen
+import androidx.core.splashscreen.SplashScreenViewProvider
+import androidx.core.view.ViewCompat
+import androidx.core.view.WindowInsetsCompat
+import androidx.core.view.postDelayed
+import androidx.interpolator.view.animation.FastOutLinearInInterpolator
+import android.transition.Transition
+import android.animation.Animator
+
+
+class MainActivity : FlutterActivity() {
+
+  var flutterUIReady : Boolean = false
+  var initialAnimationFinished : Boolean = false
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+
+    super.onCreate(savedInstanceState)
+
+    // This activity will be handling the splash screen transition
```

Comment:
nit: Period needed at the end of the sentence (here and elsewhere in this file).